webapp/WEB-INF/templates/admin/plugins/genericattributes/modules/openstreetmap/OpenStreetMapTemplate.html

```diff
@@ -1,75 +1,120 @@
+    #i18n{module.genericattributes.openstreetmap.loading}
 </div>
-<div class="form-row">
-	<div class="col col-osm osm-map" id="${entry.idEntry}_gmap"></div>
-	<#if !(editModeValue?? && editModeValue=="ReadOnly")>
-		<div id="${entry.idEntry}_addr_list" class="col col-osm list-group"></div>
-	</#if>
-</div>
-<link rel="stylesheet" href="js/plugins/genericattributes/modules/openstreetmap/leaflet.css" />
-<script src="js/plugins/genericattributes/modules/openstreetmap/leaflet.js"></script>
-<script>
-var map;
-var genericAttributes_id = '${entry.idEntry}';	
-var searchInput = "#" + genericAttributes_id + "_address";
-var xInput = "#" + genericAttributes_id + "_x";
-var yInput = "#" + genericAttributes_id + "_y";
-var resultAff = "#" + genericAttributes_id + "_addr_list";
-var addr;
-var marker = null;
+<div id="${idName}_gmap" data-map="${iterationNumber}" class="col-md-5"
+     style="height:300px; border:1px solid black;margin:5px 0;" onclick="setIdAttribute(this.id)"></div>
+<input type="hidden" name="leafletMap"/>
 <#if !(editModeValue?? && editModeValue=="ReadOnly")>
-var zoomValue=10
+    <button type="button" onclick=displayLocationOnMap(this.id) id="${idName}_gmap_button"
+            class="btn btn-primary btn-small"><i class="icon-search icon-white"></i>&nbsp;Trouver sur la carte
+    </button>
 <#else>
-var zoomValue=30
+    <#assign responseX = "" >
+    <#assign responseY = "" >
+    <#if getResponseContainingTheFieldWithCode( list_responses, "X" )?? >
+        <#assign responseX = getResponseContainingTheFieldWithCode( list_responses, "X" ).toStringValueResponse >
+    </#if>
+    <#if getResponseContainingTheFieldWithCode( list_responses, "Y" )?? >
+        <#assign responseY = getResponseContainingTheFieldWithCode( list_responses, "Y" ).toStringValueResponse >
+    </#if>
+    <#if responseX == "" || responseX == "0" || responseY == "" || responseY == "0">
+        #i18n{forms.entryType.geolocalisation.message.noGeolocation}
+    <#else>
+        <input type="hidden" name="${idName}_x" id="${idName}_x" value="${responseX!}">
+        <input type="hidden" name="${idName}_y" id="${idName}_y" value="${responseY!}">
+    </#if>
 </#if>
-$( function( ) {	
-genericAttributes_id = '${entry.idEntry}';
-var lon = '${lon!}' || $(xInput).val();
-var lat = '${lat!}' || $(yInput).val();
-bool= 'false';
-if( lat != ''  &&  lon != '' ){
-lat = lat;
-lon = lon;
-bool= 'true';
-}else {
-lat = 48.853;
-lon = 2.35;
-};
 
-map = L.map('${entry.idEntry}_gmap').setView([lat, lon], zoomValue );
-L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
-attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-}).addTo(map);
+<link rel="stylesheet" href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
+      integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY=" crossorigin=""/>
+<script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"

```

Comment:
CDN are usefull in an Internet context, but in an intranet context, it's better to include the JS library.